app/controllers/home_controller.rb

```diff
@@ -1,48 +1,45 @@
+require 'will_paginate/array'
 class HomeController < ApplicationController
   before_action :login_required
   include SamplesHelper
+
   def home
```

Comment:
i think the logic here is awfully complicated. 
You should be able to stack your query by doing. 
results = Sample.includes(:pipeline_runs, :pipeline_outputs) 
if (params[:project_id]) 
  results = results.where(project_id: 9) 
end

if (params[:search]) 
  results = results.where(name like "%%") 
end 
.....

Reply:
okay